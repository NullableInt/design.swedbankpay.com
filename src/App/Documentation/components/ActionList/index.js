import React, { Component } from "react";

import { ComponentPreview, DocToc, Icon, ExperimentalComponentAlert } from "#";
import actionList from "$/px-script/action-list";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Action lists.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="action-list">{"\n"}
                <Icon icon="more_vert" />{"\n"}
                <div className="action-menu">{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="bookmark"/>Add bookmark</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="business_center"/>Add client</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="add_circle"/>Add document</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="person_add"/>Add user</a>{"\n"}
                </div>
            </div>
        </ComponentPreview>
    </>
);

const Anchorpoints = () => (
    <>
        <h2 id="anchorpoints">Anchorpoints</h2>
        <p>Customize anchorpoint... Top right is default.</p>
        <h3>Top left</h3>
        <p>Use class <code className="token property">.anchor-top-left</code> to anchor action-list button to the top left corner of the menu.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="action-list anchor-top-left">{"\n"}
                <Icon icon="more_vert" />{"\n"}
                <div className="action-menu">{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="bookmark"/>Add bookmark</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="business_center"/>Add client</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="add_circle"/>Add document</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="person_add"/>Add user</a>{"\n"}
                </div>
            </div>
        </ComponentPreview>
        <h3>Bottom right</h3>
        <p>Use class <code className="token property">.anchor-bottom-right</code> to anchor action-list button to the bottom right corner of the menu.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="action-list anchor-bottom-right">{"\n"}
                <Icon icon="more_vert" />{"\n"}
                <div className="action-menu">{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="bookmark"/>Add bookmark</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="business_center"/>Add client</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="add_circle"/>Add document</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="person_add"/>Add user</a>{"\n"}
                </div>
            </div>
        </ComponentPreview>
        <h3>Bottom left</h3>
        <p>Use class <code className="token property">.anchor-bottom-left</code> to anchor action-list button to the bottom left corner of the menu.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="action-list anchor-bottom-left">{"\n"}
                <Icon icon="more_vert" />{"\n"}
                <div className="action-menu">{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="bookmark"/>Add bookmark</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="business_center"/>Add client</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="add_circle"/>Add document</a>{"\n"}
                    <a href="#" onClick={e => e.preventDefault()}><Icon icon="person_add"/>Add user</a>{"\n"}
                </div>
            </div>
        </ComponentPreview>
    </>
);

const ActionListText = () => (
    <div className="col-lg-10 doc-body">
        <ExperimentalComponentAlert />
        <p className="lead">Action lists...</p>
        <Overview />
        <Anchorpoints />
    </div>
);

class ActionList extends Component {
    componentDidMount () {
        actionList.init();
    }

    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <ActionListText />
                    <DocToc component={ActionListText} />
                </div>
            </div>
        );
    }
}

export default ActionList;

/* For testing */
export { Overview, Anchorpoints, ActionListText };
