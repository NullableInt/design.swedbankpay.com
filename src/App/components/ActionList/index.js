import React, { Fragment } from "react";
import PropTypes from "prop-types";

import { Icon } from "#";

const ActionList = ({ id, classNames, toggleBtn, items }) => (
    <>
        <div id={id} className="action-list">{"\n"}
            {toggleBtn ? <>{toggleBtn}{"\n"}</> : <>{<i className="material-icons action-toggle">more_vert</i>}</>}
            <div className={`action-menu${classNames ? ` ${classNames}` : ""}`}>{"\n"}
                {
                    Array.isArray(items) ? items.map(({ name, icon }, i) => (
                        <Fragment key={i}>
                            <a href="#" onClick={e => e.preventDefault()}>
                                {icon ? <Icon icon={icon}/> : null}{name}
                            </a>{"\n"}
                        </Fragment>
                    )) : null
                }
            </div>
        </div>
    </>
);

ActionList.propTypes = {
    id: PropTypes.string,
    classNames: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        icon: PropTypes.string
    })).isRequired
};

export default ActionList;
