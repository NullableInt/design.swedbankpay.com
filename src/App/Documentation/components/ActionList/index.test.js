import React from "react";
import { shallow } from "enzyme";

import ActionList, { Overview, Anchorpoints, ActionListText } from "./index";

describe("Components: ActionList", () => {
    it("is defined", () => {
        expect(ActionList).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<ActionList />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });

        it("onClick prevents default", () => {
            const wrapper = shallow(<Overview />);
            const clickHandler = { preventDefault: jest.fn() };
            const onclickAnchors = wrapper.find("a");

            onclickAnchors.forEach(anchor => anchor.simulate("click", clickHandler));
            expect(clickHandler.preventDefault).toHaveBeenCalledTimes(onclickAnchors.length);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Anchorpoints", () => {
        it("is defined", () => {
            expect(Anchorpoints).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Anchorpoints />);

            expect(wrapper).toMatchSnapshot();
        });

        it("onClick prevents default", () => {
            const wrapper = shallow(<Anchorpoints />);
            const clickHandler = { preventDefault: jest.fn() };
            const onclickAnchors = wrapper.find("a");

            onclickAnchors.forEach(anchor => anchor.simulate("click", clickHandler));
            expect(clickHandler.preventDefault).toHaveBeenCalledTimes(onclickAnchors.length);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionListText", () => {
        it("is defined", () => {
            expect(ActionListText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionListText />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
