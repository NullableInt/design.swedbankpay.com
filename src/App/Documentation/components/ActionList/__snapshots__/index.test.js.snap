// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: ActionList ActionListText renders 1`] = `
<div
  className="col-md-12 col-lg-10 doc-body"
>
  <ExperimentalComponentAlert />
  <p
    className="lead"
  >
    Action lists...
  </p>
  <Overview />
  <Anchorpoints />
</div>
`;

exports[`Components: ActionList Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Action lists.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="action-list"
    >
      

      <Icon
        icon="more_vert"
      />
      

      <div
        className="action-menu"
      >
        

        <a
          href="#"
          onClick={[Function]}
        >
          <Icon
            icon="bookmark"
          />
          Add bookmark
        </a>
        

        <a
          href="#"
          onClick={[Function]}
        >
          <Icon
            icon="business_center"
          />
          Add client
        </a>
        

        <a
          href="#"
          onClick={[Function]}
        >
          <Icon
            icon="add_circle"
          />
          Add document
        </a>
        

        <a
          href="#"
          onClick={[Function]}
        >
          <Icon
            icon="person_add"
          />
          Add user
        </a>
        

      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: ActionList renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <ActionListText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
